### PR TITLE
refactor: move the work controller to v1alpha1

### PR DESCRIPTION
**Pull request by zhiying-lin**
### Description of your changes

Will use the pkg/controllers/work for v1beta1 controller and will send out in a separate PR.

Fixes #

I have:

- [X] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

no compile errors

### CODE REVIEW

**Comment by michaelawyu** [comment]
One minor thing: should we use the organization of `work/v1alpha1`?

**Comment by zhiying-lin** [comment]
> One minor thing: should we use the organization of `work/v1alpha1`?

v1alpha1 is not a good package name. In the regular case, we will only have one controller running, and rely on the conversion webhook to serve different API versions.

Eventually, these v1alpha1 controllers will be deleted.
So prefer to have a meaningful package name.

**Comment by michaelawyu** [comment]
> > One minor thing: should we use the organization of `work/v1alpha1`?
> 
> v1alpha1 is not a good package name. In the regular case, we will only have one controller running, and rely on the conversion webhook to serve different API versions.
> 
> Eventually, these v1alpha1 controllers will be deleted. So prefer to have a meaningful package name.

The member cluster and internal member cluster package already adopts this organization I believe. And I suppose there will be a v1beta1 work controller package co-existing with the v1alpha variant, right?

**Comment by zhiying-lin** [comment]
> > > One minor thing: should we use the organization of `work/v1alpha1`?
> > 
> > 
> > v1alpha1 is not a good package name. In the regular case, we will only have one controller running, and rely on the conversion webhook to serve different API versions.
> > Eventually, these v1alpha1 controllers will be deleted. So prefer to have a meaningful package name.
> 
> The member cluster and internal member cluster package already adopts this organization I believe. Plus, there can always be package name aliases. And I suppose there will be a v1beta1 work controller package co-existing with the v1alpha variant, right?

notice that membercluster & internalmembercluster are already in that state. Probably need to refactor them later.

based on the readability rules, https://github.com/golang/go/wiki/CodeReviewComments#imports
"Avoid renaming imports except to avoid a name collision; good package names should not require renaming." Alias will be used when name collision happens. A good name could avoid that.

v1beta1 work controller will use the "work" package and v1alpha work controller is using wrong package name "controllers", which can be fixed later.

**Comment by michaelawyu** [comment]
> 

Isn't this organization a common practice in Kubernetes? We have `apps/v1`, `meta/v1`, `rbac/v1`, and many more. These practices are way more idiomatic.

File: cmd/memberagent/main.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ import (
 	fleetv1alpha1 "go.goms.io/fleet/apis/v1alpha1"
 	imcv1alpha1 "go.goms.io/fleet/pkg/controllers/internalmembercluster/v1alpha1"
 	imcv1beta1 "go.goms.io/fleet/pkg/controllers/internalmembercluster/v1beta1"
-	workapi "go.goms.io/fleet/pkg/controllers/work"
+	workapi "go.goms.io/fleet/pkg/controllers/workv1alpha1"
```
Comment (ryanzhang-oss) [inline cmd/memberagent/main.go:41]: If we have the new work controller in the work directory and move the old to workv1alpha1 then none of those changes are needed, right?

It seems that the next PR will revert all those changes

**Comment by zhiying-lin** [comment]
> > 
> 
> Isn't this organization a common practice in Kubernetes? We have `apps/v1`, `meta/v1`, `rbac/v1`, and many more. These practices are way more idiomatic and from a contextual perspective, weighs more than general guidelines.

hmm, i searched the kubernetes repo itself and v1/v1beta1 package name are only used by APIs and cannot find any controller package. I guess this kind of naming is only for the API packages?

**Comment by michaelawyu** [comment]
> > > 
> > 
> > 
> > Isn't this organization a common practice in Kubernetes? We have `apps/v1`, `meta/v1`, `rbac/v1`, and many more. These practices are way more idiomatic and from a contextual perspective, weighs more than general guidelines.
> 
> hmm, i searched the kubernetes repo itself and v1/v1beta1 package name are only used by APIs and cannot find any controller package. I guess this kind of naming is only for the API packages?

This is probably because Kubernetes updates the API they use (i.e., no two versions at the same time) -> e.g., 1.20 uses discovery v1beta1 and 1.22 discovery v1, which does not really apply to our case.

And you find tons of such organizations in, say, the `client-go` package.

File: pkg/controllers/workgenerator/controller.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ import (
 	workv1alpha1 "sigs.k8s.io/work-api/pkg/apis/v1alpha1"
 
 	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
-	workapi "go.goms.io/fleet/pkg/controllers/work"
+	workapi "go.goms.io/fleet/pkg/controllers/workv1alpha1"
```
[the remark this answers is not in the document]
Comment (zhiying-lin) [reply pkg/controllers/workgenerator/controller.go:36]: Some of the changes will be revert, cause here it does not need v1alpha1 work any more. Are you suggesting creating work v1beta1 controller and rename the old one in the PR?

**Review by michaelawyu — APPROVED** [review]
LGTM.

**Pull request merged**